lib/src/protobuf/field_info.dart

```diff
@@ -69,7 +69,7 @@ class FieldInfo<T> {
   /// Returns a read-only default value for a field.
   /// (Unlike getField, doesn't create a repeated field.)
   get readonlyDefault {
-    if (isRepeated) return const [];
+    if (isRepeated) return new List<T>();
```

Comment:
`List<Null>` is a subtype of `List<T>` for any `T`

Reply:
Not an expert in Dart so it doesn't really makes sense to me but it does seem to work. I changed it in a subsequent commit since I do prefer a const instead of a new List. Also, I tired doing `const <T>[]` but that created `List<T>` at runtime which is completely unexpected from my point of view (this caused a similar casting issue).